pgosupport.cmake

```diff
@@ -19,25 +28,57 @@ function(add_pgo TargetName)
     )
 
     # Enable PGO only for optimized configs
-    set(ConfigTypeList RELEASE RELWITHDEBINFO)
+    set(ConfigList RELEASE RELWITHDEBINFO)
+    set(IsReleaseConfig "$<OR:$<CONFIG:RELEASE>,$<CONFIG:RELWITHDEBINFO>>")
 
-    foreach(ConfigType IN LISTS ConfigTypeList)
-        set(LinkFlagsProperty "LINK_FLAGS_${ConfigType}")
-        if(CLR_CMAKE_PGO_INSTRUMENT)
+    if(CLR_CMAKE_PGO_INSTRUMENT)
+        # Unfortunately LINK_FLAGS_* don't support generator expressions, so we need to use a loop
+        foreach(Config IN LISTS ConfigList)
```

Comment:
I am sorry. I don't know why I have not taken into account the fact that it should be set for optimized builds only for Unix too. So I'd also use the `if (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE OR UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELWITHDEBINFO)` for Unix like in the other place I've suggested or use double set_property the same way as on Windows, whichever you prefer.